File: src/user-input/add-commitlint.ts
// Third-Party libraries
import inquirer from 'inquirer'

// NodeJS
import process from 'node:process'

// Utils
import { writeMessage } from '@/utils/console'
import { getErrorMessage } from '@/utils/errors'

export async function addCommitlint(): Promise<boolean> {
  try {
    const { addCommitlint } = await inquirer.prompt({
      type: 'confirm',
      default: true,
      name: 'addCommitlint',
      message: 'Do you wanna add commitlint?:'
    })

    return addCommitlint
  } catch {
    writeMessage({
      type: 'error',
      message: getErrorMessage('CommitlintSelection')
    })
    process.exit(1)
  }
}

File: internal/user-input/add-commitlint.go
package userinput

import (
	"os"

	"github.com/AlecAivazis/survey/v2"
	"github.com/RaulCatalinas/HuskyBC/internal/utils"
)

func AddCommitlint() bool {
	var questions = []*survey.Question{
		{
			Name: "addCommitlint",
			Prompt: &survey.Confirm{
				Message: "Do you wanna add commitlint?",
				Default: true,
			},
		},
	}

	answers := struct {
		AddCommitlint bool `survey:"addCommitlint"`
	}{}

	err := survey.Ask(questions, &answers)

	if err != nil {
		utils.WriteMessage(utils.WriteMessageProps{
			Type:    "error",
			Message: utils.GetErrorMessage("AddCommitLint"),
		})

		os.Exit(1)
	}

	return answers.AddCommitlint
}
